pkg/chunkenc/memchunk.go

```diff
@@ -1199,14 +1201,15 @@ func (si *bufferedIterator) moveNext() (int64, []byte, bool) {
 			return 0, nil, false
 		}
 	}
+	si.buf = si.buf[:lineSize]
+	// Take however many bytes are left in the read buffer.
+	n := copy(si.buf, si.readBuf[tWidth+lWidth:si.readBufValid])
+	// Shift down what is still left in the fixed-size read buffer, if any.
+	si.readBufValid = copy(si.readBuf[:], si.readBuf[tWidth+lWidth+n:si.readBufValid])
```

Comment:
I think this is impossible, since `si.buf` will be long enough to hold the line (logic above) and
> Copy returns the number of elements copied, which will be the minimum of len(src) and len(dst).

Adjusting `readBuf` & `readBufValid` could further complicate the next `moveNext` call if there's not enough space remaining for the varints (timestamp, linesize).

```suggestion
	si.readBufValid = 0
```

Again, I think this is impossible, but this will make the code a little clearer.

Reply:
Suppose the line was of length 0 bytes, or anything up to about 10.  Then `tWidth` is maybe 4, `lWidth` is 1 and `n` is <=10, but `readBufValid` can be 20.  So we need to copy.

Reply:
Ah, yes this makes sense. I definitely missed it on my first pass; thanks for the help and great PR!